systemvm/patches/debian/config/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -659,7 +663,10 @@ def is_guest_gateway(self, bag, ip):
         gw = interface.get_gateway()
         logging.info("Interface has the following gateway ==> %s", gw)
 
-        if bag['nw_type'] == "guest" and rip == gw:
+        guest_gw = self.config.cmdline().get_guest_gw()
+        logging.info("Interface has the following gateway ==> %s", guest_gw)
+
+        if bag['nw_type'] == "guest" and (rip == gw or rip == guest_gw):
```

Comment:
@ustcweizhou I am not sure that changes removes the need for this, but it looks like it could potentially.  Maybe I can run some tests to see if it can be removed given #1871 has been merged...